crates/core_arch/src/macros.rs

```diff
@@ -132,12 +132,37 @@ macro_rules! types {
         impl crate::fmt::Debug for $name {
             #[inline]
             fn fmt(&self, f: &mut crate::fmt::Formatter<'_>) -> crate::fmt::Result {
-                crate::core_arch::simd::debug_simd_finish(f, stringify!($name), self.as_array())
+                let array = &$crate::core_arch::macros::format_f16_as_hex!($elem_type, self.as_array());
+                crate::core_arch::simd::debug_simd_finish(f, stringify!($name), array)
             }
         }
     )*);
 }
 
+// the `intrinsic-test` crate compares the output of C and Rust intrinsics. Currently, It uses
+// a string representation of the output value to compare. In C, f16 values are currently printed
+// as hexadecimal integers. Since https://github.com/rust-lang/rust/pull/127013, rust does print
+// them as decimal floating point values. To keep the intrinsics tests working, for now, format
+// vectors containing f16 values like C prints them.
```

Comment:
f16 vector should use the same default formatting as f16. Could you change this to only apply to the Rust output of intrinsic-test instead of all users of the Debug impl?

Reply:
Right, it takes a bunch more lines but it's probably better this way.